.github/workflows/check_lint_build_release.yaml

```diff
@@ -6,17 +6,26 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  check-lint-build-release-linux:
-    name: Check, Lint, Build, Release (ubuntu stable)
+  check-lint:
+    name: Check, Lint
     runs-on: ubuntu-latest
-    permissions:
-      contents: write
-    timeout-minutes: 20
-    # env:
-    #  RUSTFLAGS: -D warnings
     steps:
       - uses: actions/checkout@v4
-      
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: rustfmt, clippy
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          key: ${{ matrix.name }}
```

Comment:
`key` is an additional per-job specifier. If you want the cache to be stable across jobs, You need to set `shared-key` and perhaps `prefix-key`. See https://github.com/Swatinem/rust-cache/tree/master

`matrix.name` is not defined here. It needs to share the cache with `ubuntu-latest`/`x86_64-unknown-linux-gnu` in `build-release`.